### ISSUE TITLE
"Copy" in code blocks inject double newlines

### ISSUE THREAD

**Issue by jippi** [squidfunk/mkdocs-material#7170]
### Context

I want to make it easier for folks to copy configuration examples from my documentation page, by using the `copy` feature.

### Bug description

When clicking the `copy` icon in all `code` blcoks on my documentation site, I get extra `blank` newlines in between the content.

I initially thought it was due to Annotations in code blocks, but does not seem to be the case.

Manually "drag cursor over text and copy" works fine.

### Related links

- [example page 1](https://jippi.github.io/scm-engine/configuration/examples/)
- [example page 2](https://jippi.github.io/scm-engine/commands/evaluate/)
- [mkdocs.yml](https://github.com/jippi/scm-engine/blob/main/mkdocs.yml)
- [Dockerfile.docs](https://github.com/jippi/scm-engine/blob/main/Dockerfile.docs)

### Reproduction

Click any `copy` icon from my `related links` section and see that you get double newlines

For example; the [`Generate labels via script`](https://jippi.github.io/scm-engine/configuration/examples/#generate-labels-via-script) code block:

## Expected 

```yaml
label:
  # Generate list of labels via script
  - strategy: generate
    # With a description (optional)
    description: "Modified this service directory"
    # With the color $pink
    color: "$pink"
    # From this script, returning a list of labels
    script: >
      /* Generate a list of files changed in the MR inside pkg/service/ */
      merge_request.modified_files_list("pkg/service/")

      /* Remove the filename from the path
        pkg/service/example/file.go => pkg/service/example */
      | map({ filepath_dir(#) })

      /* Remove the prefix "pkg/" from the path
         pkg/service/example => service/example */
      | map({ trimPrefix(#, "pkg/") })

      /* Remove duplicate values from the output */
      | uniq()
```

## Actual

```yaml
label:

  # Generate list of labels via script

  - strategy: generate

    # With a description (optional)

    description: "Modified this service directory"

    # With the color $pink

    color: "$pink"

    # From this script, returning a list of labels

    script: >

      /* Generate a list of files changed in the MR inside pkg/service/ */

      merge_request.modified_files_list("pkg/service/")



      /* Remove the filename from the path

        pkg/service/example/file.go => pkg/service/example */

      | map({ filepath_dir(#) })



      /* Remove the prefix "pkg/" from the path

         pkg/service/example => service/example */

      | map({ trimPrefix(#, "pkg/") })



      /* Remove duplicate values from the output */

      | uniq()
```

### Steps to reproduce

1. Go to any page
2. Click the `copy` icon
3. Paste into any editor or window

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [ ] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by facelessuser** [comment]
This seems to have been introduced in one of the recent Material releases. Locally, I had `mkdocs-material-9.5.6`, but after updating to ` mkdocs-material-9.5.21`, I was able to reproduce the issue.

I should note that I could only reproduce with the `line_spans: __span` option. I suspect that the copy logic gets tripped up by line spans and injects extra new lines. That is my guess at least.

**Comment by squidfunk** [comment]
Thanks for reporting. Please provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/), as requested by our [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).

**Comment by jippi** [comment]
Bisecting releases, it works as expected in `9.5.17` and fails in  `9.5.18`

**Comment by squidfunk** [comment]
Fixed in 937d7ccff103c96be0fa086cb30909bb1844b1fc. I was able to reproduce it and found out that it's related to https://github.com/squidfunk/mkdocs-material/commit/4eb1a4393741b1ecf8019dd53292b7a4d02e16f5, which fixes line highlights on mobile, but introduced the side effect of empty lines when copying.

**Comment by squidfunk** [comment]
I'm in the process of creating a release for this.

**Comment by squidfunk** [comment]
Released as part of [9.5.22](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.22).

**Comment by jippi** [comment]
Thank you @squidfunk ❤️
